spark-launch/src/main/scala/com/ibm/sparktc/sparkbench/sparklaunch/confparse/SparkJobConf.scala

```diff
@@ -0,0 +1,148 @@
+    )
+  }
+
+  def getSubmissionParams(sparkContextConf: Config): Map[String, Any] = {
+    val livyConf: Option[Map[String, Any]] =
+      Try{
+        val livyObj = sparkContextConf.getObject("livy")
+        livyObj.unwrapped().asScala.toMap
+      }.toOption
+    if(livyConf.isEmpty) {
+      val sparkHome = Try(sparkContextConf.getString("spark-home")).getOrElse(
+        getOrThrow[String](sys.env.get("SPARK_HOME"), "The environment variable SPARK_HOME must be set")
+      )
+      Map("spark-home" -> sparkHome)
+    }
+    else livyConf.get
+  }
+
+  def getSparkBenchClass(sparkContextConf: Config): String = {
+    Try(sparkContextConf.getString("class")).toOption.getOrElse("com.ibm.sparktc.sparkbench.cli.CLIKickoff")
+  }
+
+  def getSparkArgs(sparkContextConf: Config): Map[String, String] = {
+    val sparkConfMaps = Try(sparkContextConf.getObject("spark-args")).map(toStringMap).getOrElse(Map.empty)
+
+    val master: Map[String, String] = {
+      if (sparkConfMaps.contains("master")) Map.empty
+      else if(sys.env.get("SPARK_MASTER_HOST").nonEmpty) {
+        Map("master" -> sys.env("SPARK_MASTER_HOST"))
+      }
+      else Map.empty[String, String]
+    }
+    sparkConfMaps ++ master
+  }
+
+  def getSparkBenchJar(sparkContextConf: Config): String = {
+    Try(sparkContextConf.getString(SLD.sparkBenchJar))
+      .toOption
+      .getOrElse(findSparkBenchJar(sparkContextConf))
+  }
+
+  /*
+    If this condition is satisfied, then we're working from a distribution that has compiled jars.
+    We're going to just dig around in the same parent dir as this jar to find the spark-bench jar
+    because the user hasn't told us otherwise.
+  */
+  private def getDistributionJar(currentLocation: String): String = {
+    val fileList: Seq[String] = new File(currentLocation).getParentFile.listFiles().toList.map(_.getPath)
+    fileList.filterNot(_ == currentLocation).head
+  }
+
+  /*
+    Assume here that we're in a testing environment. When `sbt test` runs for spark-launch, it'll
+    first assemble the spark-bench jar and then copy it to spark-launch/test/resources/jars/spark-bench...jar.
+    We reference that folder by just "/jars" because of relative paths.
+  */
+  private def findCompiledJarInRepo(): String = {
+    val relativePath = "/jars"
+    val path = getClass.getResource(relativePath)
+    if(path == null) throw SparkBenchException("Failed to find compiled Spark-Bench jars")
+    val folder = new File(path.getPath)
+    assert(folder.exists && folder.isDirectory)
+    val filez = folder.listFiles.toList.filterNot(file => file.getName.startsWith("spark-bench-launch"))
+    filez.foreach(file => log.info(file.getName))
+    filez.filter(file => file.getName.startsWith("spark-bench")).head.getPath
+  }
+
+  def findSparkBenchJar(sparkContextConf: Config): String = {
+    val whereIAm: String = this.getClass.getProtectionDomain.getCodeSource.getLocation.getFile
+
+    if(whereIAm.endsWith(".jar")) getDistributionJar(whereIAm)
```

Comment:
you should use parentheses for all branches of this if-else block. The usual style is to only leave them out if everything fits on one line.

Reply:
Gotcha